plutus-tx-plugin/test/Plugin/Primitives/Spec.hs

```diff
@@ -156,8 +157,8 @@ stringLiteral = plc (Proxy @"stringLiteral") ("abc"::Builtins.BuiltinString)
 stringConvert :: CompiledCode (Builtins.BuiltinString)
 stringConvert = plc (Proxy @"stringConvert") ((noinline Builtins.stringToBuiltinString) "abc")
 
-stringEquals :: CompiledCode (String -> String -> Bool)
-stringEquals = plc (Proxy @"string32Equals") (\(x :: String) (y :: String) -> Builtins.equalsString (Builtins.stringToBuiltinString x) (Builtins.stringToBuiltinString y))
+stringEquals :: CompiledCode (Text -> Text -> Bool)
+stringEquals = plc (Proxy @"string32Equals") (\(x :: Text) (y :: Text) -> Builtins.equalsString (Builtins.stringToBuiltinString x) (Builtins.stringToBuiltinString y))
```

Comment:
Hmm, these tests are a bit weird. `stringToBuiltinString` should never appear in on-chain code any more, so I think you can just convert this to take `BuiltinString`s as arguments and do equality, and then construct the `BuiltinString`s directly with literals.

Reply:
Fixed.